### ISSUE TITLE
Alternate ways to get Jenkins API Token on AWS

### ISSUE THREAD

**Issue by tremulaes** [jenkins-x/jx#1316]
### Summary 

I created a new k8s cluster through jx, successfully getting to the prompt asking for the generated Jenkins API Token:

```
Jenkins X deployments ready in namespace jx


	********************************************************

	     NOTE: Your admin password is: { REDACTED }

	********************************************************

	Getting Jenkins API Token
using url http://jenkins.jx.my-website.com/me/configure
unable to automatically find API token with chromedp using URL http://jenkins.jx.my-website.com/me/configure
Please go to http://jenkins.jx.my-website.com/me/configure and click Show API Token to get your API Token
Then COPY the token and enter in into the form below:

? API Token:
```

Unfortunately, I'd just registered my DNS a short while before that and I'm stuck while waiting for the DNS to finish propagating. I think it'd be very helpful if there was another way to get this without having to use the configured DNS- if there is, it would be great to add to the documentation or messaging in this section. I think this would also be helpful for people who don't yet have their DNS setup complete.

### Steps to reproduce the behavior

Install a new k8s cluster on AWS without fully propagated DNS.

### Jx version

The output of `jx version` is:

```
NAME               VERSION
jx                 1.3.104
jenkins x platform 0.0.1827
kubernetes cluster v1.9.6
kubectl            v1.11.0
helm client        v2.9.1+g20adb27
helm server        v2.9.1+g20adb27
git                git version 2.10.1 (Apple Git-78)
```

### Kubernetes cluster

What kind of Kubernetes cluster are you using & how did you create it?

Copy-pasting my notes with sensitive data altered/redacted.

```
# Create the security group and user for kops
aws iam create-group --group-name kops
aws iam attach-group-policy --policy-arn arn:aws:iam::aws:policy/AmazonEC2FullAccess --group-name kops
aws iam attach-group-policy --policy-arn arn:aws:iam::aws:policy/AmazonRoute53FullAccess --group-name kops
aws iam attach-group-policy --policy-arn arn:aws:iam::aws:policy/AmazonS3FullAccess --group-name kops
aws iam attach-group-policy --policy-arn arn:aws:iam::aws:policy/IAMFullAccess --group-name kops
aws iam attach-group-policy --policy-arn arn:aws:iam::aws:policy/AmazonVPCFullAccess --group-name kops
aws iam create-user --user-name kops
aws iam add-user-to-group --user-name kops --group-name kops
aws iam create-access-key --user-name kops

# This should output a SecretAccessKey and AccessKeyID; record them for your records
# configure the aws client to use your new IAM user
aws configure           # Use your new access and secret key here; set region to us-east-2
aws iam list-users      # you should see a list of all your IAM users here

# Because "aws configure" doesn't export these vars for kops to use, we export them now
export AWS_ACCESS_KEY_ID=$(aws configure get aws_access_key_id)
export AWS_SECRET_ACCESS_KEY=$(aws configure get aws_secret_access_key)

# Create the state store for kops
aws s3api create-bucket --bucket kops-io --region us-east-2 --create-bucket-configuration LocationConstraint=us-east-2
aws s3api put-bucket-versioning --bucket kops-io --versioning-configuration Status=Enabled
export KOPS_STATE_STORE=s3://kops-io

jx create cluster aws --verbose \
    --zones us-east-2a,us-east-2b,us-east-2c \
    --cluster-name my-cluster \
    --git-api-token {REDACTED} \
    --git-username my-git-bot \
    --nodes 2 \
    --node-size t2.small \
    --master-size t2.small \
    --state=s3://kops-io
```

### Operating system / Environment

![image](https://user-images.githubusercontent.com/8475694/42895487-b58d2f94-8a7f-11e8-80ad-f2b1f060b74c.png)

I use zsh instead of bash.

### Expected behavior

Either:

a) a way to access the JX UI through the loadbalancer address
b) a way to retrieve the API key that doesn't require fully functional DNS.

### Actual behavior

I am unable to retrieve the Jenkins API token.

**Comment by secade** [comment]
Also just wanted to say thanks to @jstrachan for all the work here, this is a really cool project.

**Comment by rawlingsj** [comment]
I'm not sure there is a way because we've generated the ingress rules by this point.  You could leave the domain blank I guess which defaults to nip.io, then once your DNS has propagated recreate the ingress rules using the new domain for the cluster using `jx upgrade ingress --cluster`.  

If we provided sufficient docs and help during the wizard to explain this would that help?

**Comment by ruswerner** [comment]
@secade I've also hit this snag and the way I get around it is to do a `nslookup` on the Load Balancer DNS name, which should resolve right away. Then put an entry in your `/etc/hosts` file:

    35.x.x.x . jenkins.jx.<cluster>

After doing this, I was able to load the Jenkins login page, get the API token and proceed. Just remember to remove this line after the DNS propagates.

**Comment by secade** [comment]
@ruswerner Thanks for the suggestion! I can't seem to figure out how to get that to work, but my DNS should be finished propagating by tonight and I can keep myself busy until then.

@rawlingsj That's very helpful indeed. I submitted #1326 with language to help clarify the issue according to your instructions.

@jstrachan That makes a lot of sense; hopefully that update will come through soon to simplify this step.

**Comment by christian-weiss** [comment]
"jx create cluster selectedCloudHere" command should not depend on a browser at all. jx is for developers and build servers and jx should be executable in a headless server, too (without Desktop / Browser installed).

Instead of waiting for Jenkins core changes, there is maybe another way to get a user API token via ssh / shell in docker container instead of via http/https urls ...

**Comment by jstrachan** [comment]
@christian-weiss I'm afraid there's no other way to get the Jenkins API token - we'd have used it if there was. Its not Jenkins X's fault - its a limitation of Jenkins.

You can use `--headless` on jx install to use headless mode:
https://github.com/weaveworks/eksctl/issues/143

**Comment by ccojocar** [comment]
This is fixed by #2659
